opengever/maintenance/scripts/export_and_close_dossiers.py

```diff
@@ -0,0 +1,213 @@
+"""
+This script is used to export all the dossiers in a given repofolder and close
+them.
+"""
+from ftw.upgrade.progresslogger import ProgressLogger
+from opengever.document.behaviors import IBaseDocument
+from opengever.dossier.behaviors.dossier import IDossierMarker
+from opengever.dossier.interfaces import IDossierResolver
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.maintenance.debughelpers import setup_plone
+from opengever.maintenance.utils import LogFilePathFinder
+from opengever.maintenance.utils import TextTable
+from plone import api
+from zope.component import getAdapter
+import logging
+import os
+import sys
+import time
+import transaction
+
+logger = logging.getLogger('opengever.maintenance')
+logging.root.setLevel(logging.INFO)
+handler = logging.StreamHandler(stream=sys.stdout)
+formatter = logging.Formatter("%(asctime)s - %(levelname)s - %(message)s")
+handler.setFormatter(formatter)
+logging.root.addHandler(handler)
+
+TIMESTAMP = time.strftime('%d%m%Y-%H%M%S')
+
+
+class PreconditionsError(Exception):
+    """Raised when preconditions for the migration are not satisfied"""
+
+
+class DisallowedReviewState(Exception):
+    """Raised when trying to export a dossier not in an allowed final state"""
+
+
+class DossierExporter(object):
+
+    allowed_final_states = ['dossier-state-resolved', 'dossier-state-inactive']
+    states_to_resolve = ['dossier-state-active']
+
+    def __init__(self, context, output_directory, check_only=False, dryrun=False):
+        self.check_only = check_only
+        self.dryrun = dryrun
+        self.output_directory = output_directory
+        self.context = context
+        self.catalog = api.portal.get_tool("portal_catalog")
+
+    @property
+    def dossiers(self):
+        return self.catalog.unrestrictedSearchResults(
+            path=self.context.absolute_url_path(),
+            object_provides=IDossierMarker.__identifier__,
+            is_subdossier=False)
+
+    def __call__(self):
+        self.check_preconditions()
+        if self.check_only:
+            return
+
+        if self.dryrun:
+            logger.info('Performing dryrun!\n')
+            transaction.doom()
+
+        self.resolve_dossiers()
+
+        # create output directory
+        os.mkdir(self.output_directory)
+        self.export_dossiers()
+
+        if not self.dryrun:
+            logger.info("Committing...")
+            transaction.commit()
+
+        logger.info("Done!")
```

Comment:
Log output directory where the export got written to?

Reply:
Because I misread your comment, I now also log to a file in the output directory